consensus/polybft/consensus_runtime.go

```diff
@@ -512,14 +516,14 @@ func (c *consensusRuntime) calculateUptime(currentBlock *types.Header, epoch *ep
 	})
 
 	for _, addr := range addrSet {
-		uptime.addValidatorUptime(addr, uptimeCounter[addr])
+		uptime.AddValidatorUptime(addr, uptimeCounter[addr])
 	}
 
-	commitEpoch := &CommitEpoch{
-		EpochID: epoch.Number,
-		Epoch: Epoch{
-			StartBlock: epoch.FirstBlockInEpoch,
-			EndBlock:   currentBlock.Number + 1,
+	commitEpoch := &contractsapi.CommitEpochFunction{
+		ID: big.NewInt(int64(epochID)),
+		Epoch: &contractsapi.Epoch{
+			StartBlock: big.NewInt(int64(epoch.FirstBlockInEpoch)),
+			EndBlock:   big.NewInt(int64(currentBlock.Number + 1)),
```

Comment:
It is better to use SetUint64 to set `uint64`, rather than casting it to `int64` and risk potential overflow:

```suggestion
			StartBlock: new(big.Int).SetUint64(epoch.FirstBlockInEpoch),
			EndBlock:   new(big.Int).SetUint64(currentBlock.Number + 1),
```